eslint-plugin-expensify/onyx-props-must-have-default.js

```diff
@@ -0,0 +1,248 @@
+            );
+        }
+
+        // Get the node of the prop type declaration based on name
+        // For example: Component.propTypes = propTypes
+        function getPropsDeclaration(body, name) {
+            return _.find(
+                body,
+                n => lodashGet(n, 'type') === 'ExpressionStatement'
+                    && lodashGet(n, 'expression.left.property.name') === name,
+            );
+        }
+
+        // For each property making use of spread operator, get the properties from the variable
+        function getProperties(scope, propertiesVar) {
+            const resolvedProps = [];
+            _.each(
+                _.filter(
+                    propertiesVar,
+                    n => (lodashGet(n, 'type') === 'ExperimentalSpreadProperty' || lodashGet(n, 'type') === 'SpreadElement'),
+                ),
+                (prop) => {
+                    const variables = scope.variables;
+                    let variableNode = getVariableNode(variables, prop.argument.name);
+                    if (!variableNode) {
+                        // Required in case of HOCs like withPolicy, since they might have the prop types
+                        // declared outside the function, which is not in function scope
+                        variableNode = getVariableNode(scope.upper.variables, prop.argument.name);
+                    }
+
+                    if (variableNode.defs[0].type === 'ImportBinding') {
+                        // If the spread operator variable is imported
+                        const externalProps = getExternalProperties(variableNode.defs[0].node.parent.source.value, variableNode, variableNode.defs[0].node.type);
+                        resolvedProps.push(...externalProps);
+                    } else {
+                        // If the spread operator variable is in same file
+                        const internalProps = variableNode.defs[0].node.init.properties;
+                        resolvedProps.push(...internalProps);
+                    }
+                },
+            );
+            return resolvedProps;
+        }
+
+        // Get the final list of properties
+        function getPropsValue(scope, propsVar) {
+            const variables = scope.variables;
+            let props = getVariableNode(variables, propsVar);
+            if (!props) {
+                // Required in case of HOCs like withPolicy, since they might have the prop types
+                // declared outside the function, which is not in function scope
+                props = getVariableNode(scope.upper.variables, propsVar);
+            }
+
+            if (!propsVar || !props) {
+                return undefined;
+            }
+
+            // Need to check for existence of init since it is not present if the assignment is to an imported variable
+            let propsProperties;
+            if (props.defs[0].node.type === 'VariableDeclarator') {
```

Comment:
I'm not sure if `props.defs[0]` will always have value, it could be safe if we use `lodashGet`, and we also can assign `props.defs[0]` to a variable and check the value for early return.